File: backend/BudgetToWealthApi/Models/Holdings/HoldingSnapshot.cs
public class HoldingSnapshot : BaseEntity
{
    public required Guid HoldingId { get; set; }
    public Holding Holding { get; set; }
    public required DateOnly Date { get; set; }
    public required long Balance { get; set; }
    public string UserId { get; set; }
}

File: frontend/app/net-worth/holding-snapshots/components/HoldingSnapshot.ts
import { ListTableItem } from "@/app/components";
import { Holding } from "@/app/net-worth/holding-snapshots/holdings";

export type HoldingSnapshot = ListTableItem & {
  id?: number;
  holdingId: string;
  holding?: Holding;
  date: string;
  balance: number;
  userId?: string;
}

export const getHoldingSnapshotDisplayName = (snapshot: HoldingSnapshot | undefined,) => {
  return snapshot ? `${snapshot.holding?.name} - ${snapshot.holding?.institution} - ${snapshot.holding?.holdingCategory?.name} (${snapshot.holding?.type})` : '';
}